code/game/objects/structures/safe.dm

```diff
@@ -74,20 +94,63 @@ FLOOR SAFES
 			to_chat(user, "<span class='italics'>You hear a [pick("tonk", "krunk", "plunk")] from [src].</span>")
 		if(tumbler_2_pos == tumbler_2_open && turns_total == 1 ) // You cant hear tumblers if you spin fast!
 			to_chat(user, "<span class='italics'>You hear a [pick("tink", "krink", "plink")] from [src].</span>")
-	if(unlocked)
+	if(!locked)
 		if(user)
 			visible_message("<i><b>[pick("Spring", "Sprang", "Sproing", "Clunk", "Krunk")]!</b></i>")
 
 /obj/structure/safe/update_icon()
 	if(open)
-		icon_state = "[initial(icon_state)]-open"
+		if(broken)
+			icon_state = "[initial(icon_state)]-open-broken"
+		else
+			icon_state = "[initial(icon_state)]-open"
 	else
-		icon_state = initial(icon_state)
+		if(broken)
+			icon_state = "[initial(icon_state)]-broken"
+		else
+			icon_state = initial(icon_state)
+	overlays.Cut()
+	if(istype(drill, /obj/item/thermal_drill/diamond_drill))

```

Comment:
Won't this runtime if there is no drill?

Reply:
You mean if(broken)? That’s part of the structure code, just unused on the safe till now.